File: Packet/Join.cs
using System.IO;

namespace Resources.Packet {
    public class Join : Packet {
        public int unknown;
        public long guid;
        public byte[] junk;

        public Join() : base() {
            PacketID = PacketID.JoinPacket;
        }
        public Join(BinaryReader reader) : this() {
            unknown = reader.ReadInt32();
            guid = reader.ReadInt64();
            junk = reader.ReadBytes(0x1168);
        }

        protected override void WritePacketData(BinaryWriter writer) {
            writer.Write(unknown);
            writer.Write(guid);
            writer.Write(junk);
        }
    }
}

File: src/main/kotlin/packets/Join.kt
package com.github.lastexceed.cubeworldnetworking.packets

import com.github.lastexceed.cubeworldnetworking.utils.*

class Join(
	val unknown: Int,
	val assignedID: CreatureId,
	val junk: ByteArray
) : Packet(PacketId.Join) {
	override suspend fun writeTo(writer: Writer) {
		writer.writeInt(unknown)
		writer.writeLong(assignedID.value)
		writer.writeByteArray(junk)
	}

	companion object : CwDeserializer<Join> {
		override suspend fun readFrom(reader: Reader) =
			Join(
				unknown = reader.readInt(),
				assignedID = CreatureId(reader.readLong()),
				junk = reader.readByteArray(0x1168)
			)
	}
}